pkg/config/config.go

```diff
@@ -251,8 +260,15 @@ type Config struct {
 	// not enabled. Defaults to "http".
 	DefaultExternalScheme string
 
-	// DefaultExternal specifies whether internal traffic is encrypted or not.
+	// Deprecated - replaced with InternalDataplaneTrust and InternalControlplaneTrust
+	// InternalEncryption specifies whether internal traffic is encrypted or not.
 	InternalEncryption bool
+
+	// InternalDataplaneTrust specifies the level of trust used for date plane.
+	InternalDataplaneTrust string
```

Comment:
Since the values will be referred by other repos, should the values `disabled`, `minimum`, `enabled`, ...etc be introduced as new type like HTTPProtocol https://github.com/knative/networking/blob/68725bdd1056db9dcf7f35b1ea192198045b4351/pkg/config/config.go#L132 ?